monitor-function-calls/src/utils.js

```diff
@@ -0,0 +1,171 @@
+const BigNumber = require('bignumber.js');
+
+function isNumeric(valueString) {
+  // Check the substrings for a valid numeric expression
+  // (characters 0-9) (optional decimal) (optional characters 0-9)
+  const result = valueString.match(/^[0-9]*?[.]?[0-9]*$/);
+  if (result === null) {
+    return false;
+  }
+  return (result[0].length === result.input.length);
+}
+
+function isAddress(valueString) {
+  const result = valueString.match(/^0x[0-9a-f]{40}$/);
+  if (result === null) {
+    return false;
+  }
+  return (result[0].length === result.input.length);
+}
+
+function addressComparison(variable, operator, operand) {
+  switch (operator) {
+    case '===':
+      return variable.toLowerCase() === operand.toLowerCase();
+    case '!==':
+      return variable.toLowerCase() !== operand.toLowerCase();
+    default:
+      throw new Error(`Address operator ${operator} not supported`);
+  }
+}
+
+function booleanComparison(variable, operator, operand) {
+  switch (operator) {
+    case '===':
+      return variable === operand;
+    case '!==':
+      return variable !== operand;
+    default:
+      throw new Error(`Boolean operator ${operator} not supported`);
+  }
+}
+
+function bigNumberComparison(variable, operator, operand) {
+  switch (operator) {
+    case '===':
+      return variable === operand;
```

Comment:
Does this work for BigNumbers? BigNumbers have an `eq` method that can be used instead